### PR TITLE
[BD-29] [TNL-6993] [BB-2706] Make Poll and Survey Indexable

### PR DESCRIPTION
**Pull request by ahmedbilal**
Make Poll and Survey Indexable

**JIRA tickets**: 
* https://openedx.atlassian.net/browse/TNL-6993

**Merge deadline**: "None"

**Testing instructions**:

1. Clone this repo in `src` directory of your devstacks.
2. Checkout this branch i.e `ahmed/bb-2706-make-poll-and-survey-indexable`
3. Install the cloned project into both **LMS** and **Studio** by changing your directory to the `src/xblock-poll` directory in your devstack and executing `pip install -e .`
4. Add `poll` and `survey` to the Advanced Module List. See https://github.com/open-craft/xblock-poll#setting-up-a-course-to-use-polls-and-surveys
5. Create two blocks. One poll and other survey.
6. Reindex your course.
7. Go to LMS and search for some string that appears in the poll or survey

**Reviewers**
- [ ] @xitij2000 
- [ ] @davestgermain

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def package_data(pkg, roots):
         'markdown',
         'ddt',
         'mock',
+        'bleach==3.1.5',
```
Comment (xitij2000) [inline setup.py:59]: Don't pin the version here.

File: poll/poll.py [hunk 2]
```diff
@@ -840,6 +840,41 @@ def generate_report_data(self, user_state_iterator, limit_responses=None):
...
+    def index_dictionary(self):
+        """
+        Return dictionary prepared with module content and type for indexing.
+        """
+        # return key/value fields in a Python dict object
+        # values may be numeric / string or dict
+        # default implementation is an empty dict
+        xblock_body = super(PollBlock, self).index_dictionary()
+        answers = {
+            "answer_{}".format(answer_i): remove_markdown_and_html_tags(answer[1]['label'])
```
Comment (xitij2000) [inline poll/poll.py:852]: Probably not that relevant, but given that this is a poll, these aren't answers, but options. 

Same for surveys below.

**Comment by ahmedbilal** [comment]
@xitij2000 Can you take a look at failed CI error? I don't think it is related.

**Comment by xitij2000** [comment]
@ahmedbilal You can always log into the CI and rerun the failed tests. I've done so and merged the PR.

**Comment by ahmedbilal** [comment]
@xitij2000 I didn't know that. Thanks for tip :+1:

**Comment by lgp171188** [comment]
@ahmedbilal, @xitij2000, since a new tag/release was not created after this change was merged, I have requested @shimulch to merge https://github.com/open-craft/xblock-poll/pull/83 with the same version and tag a new release `0.10.0` including that change as well. FYI.

**Pull request merged**